### PR TITLE
solved sqrt(1+sin(x))

### LINKED ISSUE #15730: Unable to integrate sqrt(1+sin(x))
`from sympy import *`
`x, y, z, t = symbols('x y z t') `

Now 

`integrate(sqrt(1+sin(x)),x)` gives me back what I put in.

### PR DESCRIPTION
**Pull request by RituRajSingh878**
#### References to other Issues or PRs

Fixes #15730.

#### Brief description of what is fixed or changed
unable to integrate integrate(sqrt(1+sin(x)),x) ,so i add a case of this function in manual integrate .

#### Release Notes

- integrals
  - added (sqrt(1+sin(x)),x)  case

### CODE REVIEW

**Comment by oscarbenjamin** [comment]
This patch looks very specific. It seems to only solve this exact integral. Is it possible to make it more general than this?

The other cases above in the code seem to handle classes of integrals e.g. every integral of the form `integrate(csc(f(x)), x)` for some function `f(x)`. Can you take the idea of how you've solved this integral and turn it into something that can solve a class of integrals?

You should add tests for any new integrals that can evaluated.

**Comment by RituRajSingh878** [comment]
> This patch looks very specific. It seems to only solve this exact integral. Is it possible to make it more general than this?
I am trying ,if you can give some suggestion it will be very helpful .

**Comment by RituRajSingh878** [comment]
> The simplest generalisation would be if there were some constants like `a*sqrt(1+sin(b*x))` but you can probably do something more general. I'm going away for a while now so I can't help more I'm afraid...

I am trying to change fu.py so that I can make it more general.if from fu.py I can remove sympy.sqrt from expression, then it will be very easy to integrate it,

**Comment by oscarbenjamin** [comment]
Why are you trying to change fu.py? What general rule are you trying to implement?

**Comment by RituRajSingh878** [comment]
> Why are you trying to change fu.py? What general rule are you trying to implement?

sqrt(1+sin(x)) can be change into sqrt(1+2*si(x/2)*cos(x/2)) by TR11 rule  in fu.py , so i want to add some rule such that it can change it  into |sin(x/2)+cos(x/2)|.Then we can integrate it eaisly.

**Comment by RituRajSingh878** [comment]
Here my main  intention is not only to solve integration but I also want to simplify sqrt(a+a*sin(x)) or sqrt(a+a*cos(x)) so from their we can possibly solve these integration.
and TR11(sqrt(1+sin(x)),base=x/2) is giving sqrt(1+2*sin(x/2)*cos(x/2)) so I just want to change it into |sin(x/2)+cos(x/2)| so from here we have to not add any method for integration.

File: sympy/integrals/manualintegrate.py [hunk 1]
```diff
@@ -696,8 +697,13 @@ def trig_rule(integral):
...
         rewritten = ((sympy.csc(arg)**2 + sympy.cot(arg) * sympy.csc(arg)) /
                      (sympy.csc(arg) + sympy.cot(arg)))
+    elif integrand == sympy.sqrt(1+sympy.sin(symbol)):
+        rewritten = (sympy.sin(symbol/2) + sympy.cos(symbol/2))
     else:
-        return
+        match = integrand.match(sympy.sqrt(1 + sympy.sin(a*symbol)))
+        if not match:
+            return
+        rewritten = (sympy.sin((match[a]*symbol)/2) + sympy.cos((match[a]*symbol)/2))
```
Comment (oscarbenjamin) [inline sympy/integrals/manualintegrate.py]: This case is equivalent to the one above if a=1 so we don't really need two cases: this code can be simplified

Comment (RituRajSingh878) [reply sympy/integrals/manualintegrate.py]: Ooohhh I just forgot to remove the above code.

**Comment by oscarbenjamin** [comment]
Okay this looks better. Can you add some tests for this?

**Comment by oscarbenjamin** [comment]
Is there a minus sign missing somewhere? Why am I getting this (from the test)?
```julia
In [29]:     f = sqrt(5+5*sin(5*x))
    ...:     F = sqrt(5)*(2*sin(5*x/2)/5-2*cos(5*x/2)/5)
 
In [25]: F.diff(x).subs(x, 1)
Out[25]: √5⋅(cos(5/2) + sin(5/2))

In [26]: f.subs(x, 1)
Out[26]: 
  ______________
╲╱ 5⋅sin(5) + 5 

In [27]: F.diff(x).subs(x, 1).evalf()
Out[27]: -0.453187187246404

In [28]: f.subs(x, 1).evalf()
Out[28]: 0.453187187246404
```

**Comment by RituRajSingh878** [comment]
The main problem is with sqrt(5 +5*sin(5*x)).It should change into `sqrt(5)*abs(sin(5*x/2)-cos(5*x/2))` .

eg. `x=Symbol("x",integer=True), sqrt(x**2) = abs(x)` and open like abs(x) ={x: x>=0,- x :x<0}

**Comment by oscarbenjamin** [comment]
Why not put `abs()` in the rewrite expression then? Or does that mean that the integral doesn't work any more?

I notice that wolfram gives a [more complicated answer](https://www.wolframalpha.com/input/?i=integrate+sqrt(5%2B5*sin(5*x))). SymPy confirms that answer:
```julia
In [27]: f = sqrt(5+5*sin(5*x))                                                                                                                

In [28]: F = 2*sqrt(sin(5*x) + 1)* (sin((5*x)/2) - cos((5*x)/2))/(sqrt(5)*(sin((5*x)/2) + cos((5*x)/2)))                                       

In [29]: simplify(F.diff(x)-f)                                                                                                                 
Out[29]: 0
```

**Comment by oscarbenjamin** [comment]
It looks like you've removed the test. This will need a (correct) test to be accepted.

**Comment by RituRajSingh878** [comment]
@asmeurer can you please check it.

**Comment by RituRajSingh878** [comment]
@jksuom  can you check this also?

**Comment by oscarbenjamin** [comment]
Your test is that `f` integrates to `F` where
```julia
In [12]: f = sqrt(1+sin(x))                                                                                                       

In [13]: f                                                                                                                        
Out[13]: 
  ____________
╲╱ sin(x) + 1 

In [14]:     F = Piecewise((2*sqrt(-sin(x) + 1), x <= 0), (-2*sqrt(-sin(x) + 1) + 4, x <= pi/2),\ 
    ...:     (2*sqrt(-sin(x) + 1) + 4, x <= 3*pi/2), (-2*sqrt(-sin(x) + 1) + 4 + 4*sqrt(2), x <= 2*pi),\ 
    ...:     (2*sqrt(-sin(x) + 1) + 4*sqrt(2), True))                                                                             

In [15]: F                                                                                                                        
Out[15]: 
⎧          _____________                    
⎪      2⋅╲╱ -sin(x) + 1           for x ≤ 0 
⎪                                           
⎪         _____________                   π 
⎪   - 2⋅╲╱ -sin(x) + 1  + 4       for x ≤ ─ 
⎪                                         2 
⎪                                           
⎪        _____________                   3⋅π
⎨    2⋅╲╱ -sin(x) + 1  + 4       for x ≤ ───
⎪                                         2 
⎪                                           
⎪      _____________                        
⎪- 2⋅╲╱ -sin(x) + 1  + 4 + 4⋅√2  for x ≤ 2⋅π
⎪                                           
⎪       _____________                       
⎪   2⋅╲╱ -sin(x) + 1  + 4⋅√2      otherwise 
⎩   
```

Comparing this by differentiation gives:
```julia
In [16]: simplify(F.diff(x)-f)                                                                                                    
Out[16]: 
⎧    ____________        cos(x)                 
⎪- ╲╱ sin(x) + 1  - ───────────────   for x ≤ 0 
⎪                     _____________             
⎪                   ╲╱ -sin(x) + 1              
⎪                                               
⎪    ____________        cos(x)               π 
⎪- ╲╱ sin(x) + 1  + ───────────────   for x ≤ ─ 
⎪                     _____________           2 
⎪                   ╲╱ -sin(x) + 1              
⎪                                               
⎪    ____________        cos(x)              3⋅π
⎪- ╲╱ sin(x) + 1  - ───────────────  for x ≤ ───
⎨                     _____________           2 
⎪                   ╲╱ -sin(x) + 1              
⎪                                               
⎪    ____________        cos(x)                 
⎪- ╲╱ sin(x) + 1  + ───────────────  for x ≤ 2⋅π
⎪                     _____________             
⎪                   ╲╱ -sin(x) + 1              
⎪                                               
⎪    ____________        cos(x)                 
⎪- ╲╱ sin(x) + 1  - ───────────────   otherwise 
⎪                     _____________             
⎪                   ╲╱ -sin(x) + 1              
⎩                                               
```
If the answer provided is correct then this should be equivalent to zero somehow. I don't think it is though.  I think the cases with `+` can be simplified to zero (perhaps depending on assumptions) but those with `-` can not.

Ideally we'd have a way of getting the answer the Wolfram provides which is not piecewise and doesn't seem to need assumptions on `x`:
```julia
In [34]: x = Symbol('x')                                                                                                          

In [35]: F = (2 * sqrt(sin(x) + 1) * (sin(x/2) - cos(x/2)))/(sin(x/2) + cos(x/2))                                                 

In [36]: F                                                                                                                        
Out[36]: 
  ⎛   ⎛x⎞      ⎛x⎞⎞   ____________
2⋅⎜sin⎜─⎟ - cos⎜─⎟⎟⋅╲╱ sin(x) + 1 
  ⎝   ⎝2⎠      ⎝2⎠⎠               
──────────────────────────────────
            ⎛x⎞      ⎛x⎞          
         sin⎜─⎟ + cos⎜─⎟          
            ⎝2⎠      ⎝2⎠          

In [37]: simplify(F.diff(x)-sqrt(1+sin(x)))                                                                                       
Out[37]: 0
```

**Comment by jksuom** [comment]
`sqrt(1 + sin(x))` has two entire analytic branches, `sin(x/2) + cos(x/2)` and `-sin(x/2) - cos(x/2)`, which are periodic of period `4*pi`. On the real line, the former is positive (and hence equal to the principal branch of `sqrt(1 + sin(x))`) in the interval `(-pi/2, 3*pi/2)` and all of its translates by a multiple of `4*pi`. The latter branch is positive on the remaining intervals. This means that we can write

    sqrt(1 + sin(x)) = sgn(x)*(sin(x/2) + cos(x/2))

 where `sgn(x) = sqrt(1 + sin(x))/(sin(x/2) + cos(x/2)) is equal to 1 or -1 on alternate intervals.

Wolfram seems to work by integrating piecewise the non-constant terms `sin(x/2) + cos(x/2)` thus obtaining `sgn(x)*2*(-cos(x/2) + sin(x/2)`. However, this function is not the correct antiderivative. It is discontinuous at the endpoints of the intervals. (The antiderivative of a non-negative function cannot be periodic.)

**Comment by oscarbenjamin** [comment]
Okay that all makes sense. So the function `f` can be rewritten and integrated as
```julia
In [54]: f = Piecewise((sin(x/2) + cos(x/2), sin(x/2) + cos(x/2) >= 0), (-sin(x/2)-cos(x/2), True))                               

In [55]: f                                                                                                                        
Out[55]: 
⎧    ⎛x⎞      ⎛x⎞          ⎛x⎞      ⎛x⎞    
⎪ sin⎜─⎟ + cos⎜─⎟   for sin⎜─⎟ + cos⎜─⎟ ≥ 0
⎪    ⎝2⎠      ⎝2⎠          ⎝2⎠      ⎝2⎠    
⎨                                          
⎪     ⎛x⎞      ⎛x⎞                         
⎪- sin⎜─⎟ - cos⎜─⎟         otherwise       
⎩     ⎝2⎠      ⎝2⎠                         

In [56]: integrate(f)                                                                                                             
Out[56]: 
⎧            ⎛x⎞        ⎛x⎞                   
⎪     - 2⋅sin⎜─⎟ + 2⋅cos⎜─⎟         for x ≤ 0 
⎪            ⎝2⎠        ⎝2⎠                   
⎪                                             
⎪         ⎛x⎞        ⎛x⎞                   3⋅π
⎪    2⋅sin⎜─⎟ - 2⋅cos⎜─⎟ + 4       for x ≤ ───
⎪         ⎝2⎠        ⎝2⎠                    2 
⎪                                             
⎪       ⎛x⎞        ⎛x⎞                     7⋅π
⎨- 2⋅sin⎜─⎟ + 2⋅cos⎜─⎟ + 4 + 4⋅√2  for x ≤ ───
⎪       ⎝2⎠        ⎝2⎠                      2 
⎪                                             
⎪      ⎛x⎞        ⎛x⎞                         
⎪ 2⋅sin⎜─⎟ - 2⋅cos⎜─⎟ + 4 + 8⋅√2   for x ≤ 4⋅π
⎪      ⎝2⎠        ⎝2⎠                         
⎪                                             
⎪         ⎛x⎞        ⎛x⎞                      
⎪  - 2⋅sin⎜─⎟ + 2⋅cos⎜─⎟ + 8⋅√2     otherwise 
⎩         ⎝2⎠        ⎝2⎠                      
```
which is correct only for `0 <= x <= 4*pi`.

**Comment by RituRajSingh878** [comment]
> which is correct only for `0 <= x <= 4*pi`.

f is periodic so it will work in` (4*n*pi,4*n*pi+4*pi)`

**Comment by oscarbenjamin** [comment]
> f is periodic so it will work in` (4*n*pi,4*n*pi+4*pi)` interval

f is periodic but it's integral over a period is non-zero so the integral of F is not periodic. `integrate` there has given an expression for `x > 4` that is incorrect.

**Comment by czgdp1807** [comment]
@RituRajSingh878 Are you still interested to work on this?

**Comment by RituRajSingh878** [comment]
you can if you wish

**Comment by czgdp1807** [comment]
@jksuom Can we close this PR, since the associated issue is still open, so someone interested can work on it.

**Comment by jksuom** [comment]
I think it can be closed. More work is needed for integrals of periodic functions.

**Pull request closed without merging**